components/playlist/browser/resources/reducers/playlist_reducer.ts

```diff
@@ -20,10 +22,21 @@ const playlistReducer: Reducer<PlaylistData | undefined> = (
 
   switch (action.type) {
     case types.PLAYLIST_LOADED:
-      const playlists = action.payload
-      let currentList: PlaylistMojo.Playlist | undefined
+      const playlists = action.payload as Playlist[]
+
+      // Fix up local thumbnail url to chrome-untrusted
+      if (location.protocol === 'chrome-untrusted:') {
+        for (const playlist of playlists) {
+          for (const item of playlist.items) {
+            fixUpThumbnailURL(item)
+            fixUpMediaURL(item)
+          }
+        }
+      }
+
+      let currentList: Playlist | undefined
       if (state.currentList) {
-        currentList = playlists.find((list: PlaylistMojo.Playlist) => {
+        currentList = playlists.find((list: Playlist) => {
           return list.id === state?.currentList?.id
```

Comment:
State can never be undefined, I think, because of line 20
```suggestion
          return list.id === state.currentList?.id
```

Reply:
Yeah, I agree. Unfortunately, my local dev env seems to complain it could be undefined, which is a false alarm.

Reply:
Ah right, the makes sense :laughing: sorry for the noise!